### PR TITLE
7935 - Clicking Prefix/Suffix in TextField does not result in the focus being set.

### LINKED ISSUE #7935: Clicking Prefix/Suffix in TextField does not result in the focus being set.
### Environment Information 
 
- __Package version(s)__: current
- __Browser and OS versions__: all

#### Actual behavior:
 

Clicking the prefix or suffix section does nothing.
 
#### Expected behavior:
Clicking the prefix or suffix section of a TextField should result in the TextField receiving focus.

#### Priorities and help requested:
 
Are you willing to submit a PR to fix? Yes
 
Requested priority: Low
 
Products/sites affected: (if applicable)

### PR DESCRIPTION
**Pull request by redoz**
#### Pull request checklist

- [X] Addresses an existing issue: Fixes #7935 
- [X] Include a change request file using `$ npm run change`

#### Description of changes
This will ensure clicking the prefix/suffix area of a textbox results in the input field receiving focus.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/7937)

### CODE REVIEW

**Comment by redoz** [comment]
I was trying to rush this out before EOD but it went really wrong, I'll see if I can fix it tonight, otherwise I'll probably pick this up on Monday.

**Comment by kenotron** [comment]
Before you implement too much more code, @redoz, was wondering if you could just accomplish this stuff with `<label>` to do what you're trying to do with JS...

**Comment by redoz** [comment]
@kenotron That's a good idea, I'll give it a try. My initial change was minimal though, so not a lot effort went to waste.

**Comment by redoz** [comment]
@kenotron Would you suggest I turn the wrapping div into the label, or the default-rendered inner span?

**Comment by redoz** [comment]
@kenotron I think this might be too much HTML/CSS for me to fix, I think the only thing worth salvaging are the Screener tests (and even those might not be worth keeping). 

Should I just close this PR and let someone else take have a go at it?

**Comment by kenotron** [comment]
@redoz - don't get discouraged! what you have seems "okay" but it does change structure of the HTML generated. This can affect products, etc. I'm a bit hazy on how labels & input fields are interpreted by Narrator. So that might be my biggest concern. I was actually thinking that you would have swapped divs with labels here. We can definitely use the VR tests tho!

@dzearing - would you classify changing a `div` to `label` here to be a major breaking change?

**Comment by redoz** [comment]
@kenotron I'm not discouraged per se, but I am not a front-end developer so once something turns into a DOM/CSS issue I usually defer to someone who is. I thought it would be a quick one-line fix, and now it's not, so it might be better to leave this one for someone who actually knows what they are doing :)

**Comment by redoz** [comment]
That turned out better than expected, I'll let someone with more understanding of HTML decide whether this is an acceptable fix though.

**Comment by redoz** [comment]
I noticed that clicking the border does not result in the input receiving focus, which I would argue is a bug as it's already in the 'hover' state. But I'm considering it out of scope for this PR.

**Comment by kenotron** [comment]
I think that's okay. @micahgodbolt - do you feel okay with the div -> label swap? I might have to do a wider email when this is checked in to let people know maybe.

File: apps/vr-tests/src/stories/TextField.stories.tsx [hunk 1]
```diff
@@ -49,3 +49,35 @@ storiesOf('TextField', module)
       rtl: true
     }
   );
+
```
Comment (kenotron) [inline apps/vr-tests/src/stories/TextField.stories.tsx]: I don't think this screener test is actually necessary. There is already a suffix / prefix test to catch the style of things. For a real test that would catch focus, you need to do enzyme testing.

Comment (redoz) [reply apps/vr-tests/src/stories/TextField.stories.tsx]: I'm afraid I know nothing of enzyme, the intent of this test was that it clicks the prefix/suffix so you can verify it ended up in a focused state. But like I said, if it's not worth keeping feel free to throw it out, I don't mind.

Comment (kenotron) [reply apps/vr-tests/src/stories/TextField.stories.tsx]: Yeah, go ahead and delete that test for now. I think we can backfill with some tests if necessary. That will get you to pass the tests, etc.

Comment (redoz) [reply apps/vr-tests/src/stories/TextField.stories.tsx]: Sure, I'll trim it out tomorrow.

**Comment by micahgodbolt** [comment]
@kenotron @redoz I think this is certainly worth a wider audience. I'm on the fence as to if those prefix/sufixs should be labels or not AND we should certainly determine if they should be read by screen readers (which they are currently not). I think once we know the proper semantics here, we can apply a solution.

**Comment by redoz** [comment]
@micahgodbolt Shall I just close this then?

**Comment by micahgodbolt** [comment]
@redoz Yeah, lets figure out how we want to tackle this issue first, then open a PR. Appreciate your flexibility!

**Pull request closed without merging**